hcl2template/shim/values.go

```diff
@@ -217,3 +222,70 @@ func WriteUnknownPlaceholderValues(v cty.Value) cty.Value {
 		panic("unknown type")
 	}
 }
+
+func HCL2ValueFromConfig(conf interface{}, configSpec map[string]hcldec.Spec) cty.Value {
+	c := map[string]interface{}{}
+	if err := mapstructure.Decode(conf, &c); err != nil {
+		panic(fmt.Errorf("can't convert %#v to cty.Value", conf))
```

Comment:
Yup, panicing sounds harsh but this should only happen in dev mode but I think we should add a comment here for the plugin building people 🙂 .

Reply:
Add a comment and still panic when this fails?